### ISSUE TITLE
Binaryen can't parse all spec wasts

### ISSUE THREAD

**Issue by tlively** [WebAssembly/binaryen#1138]
In particular, it can't parse test/spec/nop.wast
```
$ bin/wasm-opt --print test/spec/nop.wast
[parse exception: expected list (at 7:4)]Fatal: error in parsing input
```
This is because of the non-sexp format used in the `3-ary` function.

**Comment by kripken** [comment]
Yeah, would be nice to fix this. We don't parse the stacky non-sexp format that wasm changed to, binaryen's text format is basically the previous s-exp format. It would be nice to have a parser able to handle both.

**Comment by tlively** [comment]
We still can't parse all the spec tests, but that's no longer because of the overall architecture of the parser, so I'll go ahead and close this.
